### PR TITLE
conformance test: tear down manifests before blobs

### PR DESCRIPTION
**Pull request by majewsky**
As explained in the added comments, "some registries" (i.e. my own implementation, Keppel <https://github.com/sapcc/keppel>) forbid deleting blobs that are still referenced by manifests. To do a proper cleanup in this case, the manifest needs to be deleted first.

The diff looks a bit weird, but I really just swapped the order of some paragraphs, pulling the manifest deletion to the top.

### CODE REVIEW

**Comment by majewsky** [comment]
> This PR should be modified to switch the order based on the presence of some env var

Done.

**Comment by majewsky** [comment]
Documentation added.

File: conformance/01_pull_test.go [hunk 1]
```diff
@@ -152,6 +153,21 @@ var test01Pull = func() {
 		})
 
 		g.Context("Teardown", func() {
+			deleteManifestFirst, _ := strconv.ParseBool(os.Getenv(envVarDeleteManifestBeforeBlobs))
```
Comment (jdolitsky) [inline conformance/01_pull_test.go:156]: looking through this one more time - could we move this env var check to run only once at the bottom of setup.go? Check something like `runContentDiscoverySetup` , it is initialized as a global then set based on env

Comment (majewsky) [reply conformance/01_pull_test.go]: done

**Comment by jdolitsky** [comment]
@majewsky looks good. Last thing is the DCO check failed on your last commit. Can you amend the commit to contain signoff then we can merge?

**Comment by jdolitsky** [comment]
@majewsky - thanks for you patience, looks like the final commit had a different email in signed-off-by which might be causing an error: https://pullapprove.com/opencontainers/distribution-spec/pull-request/156/requirements/

**Comment by jzelinskie** [comment]
LGTM

**Pull request closed without merging**